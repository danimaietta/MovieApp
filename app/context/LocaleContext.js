import React from 'react'

const LocaleContext = React.createContext({
  theme: 'light',
  setTheme: () => (theme === 'light' ? 'dark' : 'light')
})

export default LocaleContext
