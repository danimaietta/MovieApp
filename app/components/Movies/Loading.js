import React, { useContext } from 'react'
import LocaleContext from '../../context/LocaleContext'

export default function Loading() {
  const { theme } = useContext(LocaleContext)
  const classBtn = theme == 'light' ? 'light-button' : 'dark-button'

  console.count('Loading')

  return (
    <div className=' container95 flex y-center center'>
      <h2 className={`${classBtn}`}>Loading...</h2>
    </div>
  )
}
