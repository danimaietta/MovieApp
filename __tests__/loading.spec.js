import React from 'react'
import Loading from '../app/components/Movies/Loading'
import LocaleContext from '../app/context/LocaleContext'
import {
  render,
  screen,
  fireEvent,
  queryByTestId,
  queryByPlaceholderText
} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

const theme = 'dark'

describe('Loading', () => {
  test('Check if displays Loading... message ', () => {
    const { getByText } = render(
      <LocaleContext.Provider value={{ theme }}>
        <Loading />
      </LocaleContext.Provider>
    )
    expect(getByText('Loading...')).toBeInTheDocument()
  })
  test('Check if displays Loading... message ', () => {
    const { container } = render(
      <LocaleContext.Provider value={{ theme }}>
        <Loading />
      </LocaleContext.Provider>
    )
    expect(container.firstChild).toMatchSnapshot(`
    <div
       class=" container95 flex y-center center"
     >
       <h2
         class="dark-button"
       >
         Loading...
       </h2>
     </div>
    `)
  })
})
