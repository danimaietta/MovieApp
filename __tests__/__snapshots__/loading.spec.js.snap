// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Check if displays Loading... message : 
    <div
       class=" container95 flex y-center center"
     >
       <h2
         class="dark-button"
       >
         Loading...
       </h2>
     </div>
     1`] = `
<div
  class=" container95 flex y-center center"
>
  <h2
    class="dark-button"
  >
    Loading...
  </h2>
</div>
`;
